src/biome/text/api_new/modules/heads/bimpm_classification.py

```diff
@@ -0,0 +1,487 @@
+from typing import Optional, Dict, List, Any
+
+import torch
+from allennlp.common.checks import check_dimensions_match
+from allennlp.data import Instance
+from allennlp.modules import (
+    FeedForward,
+    Seq2SeqEncoder,
+    Seq2VecEncoder,
+    TimeDistributed,
+)
+from allennlp.modules.bimpm_matching import BiMpmMatching
+from allennlp.nn import InitializerApplicator
+from allennlp.nn import util
+from biome.text.api_new.model import Model
+from biome.text.api_new.modules.encoders import TimeDistributedEncoder
+from biome.text.api_new.modules.heads.classification.defs import ClassificationHead
+from overrides import overrides
+
+
+class BiMpm(ClassificationHead):
```

Comment:
Should be nice implement the `prediction_explain` for enable explains functionality in this kind of model.

You can take a look into [`DocumentClassification`](https://github.com/recognai/biome-text/blob/feat/add_our_biome_bimpm/src/biome/text/api_new/modules/heads/doc_classification.py#L103) and [`TextClassification`](https://github.com/recognai/biome-text/blob/feat/add_our_biome_bimpm/src/biome/text/api_new/modules/heads/text_classification.py#L73) implementations, since should be a merge of both

Reply:
agreed, would be very nice indeed! I will have a look at it tomorrow.